src/ruleset.hpp

```diff
@@ -169,7 +172,7 @@ struct ruleset {
     std::shared_ptr<action_mapper> actions;
 
     // The key used to organise collections is rule.type
-    std::unordered_set<std::string_view> collection_types;
+    std::unordered_set<std::string> collection_types;
```

Comment:
I'd have a slight preference see some struct along the lines of

```c++
struct collection_key {
    std::string_view type;
    std::string_view module;
}
```

with operator== and hash function, but if you're going this route you should update the comment above.

Reply:
As mentioned offline, this is a fair point but being a quick fix and not something I plan to keep for long, I'll just update the comment.